### ISSUE TITLE
X-Forwarded-Host seems wrong on frontends configured by file backend

### ISSUE THREAD

**Issue by melchor629** [containous/traefik#3694]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

I have configured some docker containers to work with traefik, and one file entry (config file below) for a Jenkins not in a container. Jenkins keep saying that the "inverse proxy seems to be wrongly configured", so I decided to catch some network traffic and see HTTP headers. I found that the `X-Forwarded-Host` on the docker containers are the expected ones, but in the frontend of the file entry, that header, is incorrect.

Here is an example of request headers made from traefik to one of the containers (request is `http://pi.local`):

```
	GET / HTTP/1.1
	Host: pi.local
	User-Agent: ...
	Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
	Accept-Encoding: gzip, deflate
	Accept-Language: es-es
	Cookie: jenkins-timestamper-offset=-7200000
	Upgrade-Insecure-Requests: 1
	X-Forwarded-For: x.x.x.x
	X-Forwarded-Host: pi.casita.local
	X-Forwarded-Port: 80
	X-Forwarded-Proto: http
	X-Forwarded-Server: eb2a2f09a409
	X-Real-Ip: x.x.x.x
```

The `X-Forwarded-Host` is the expected `pi.local`.

Here is another example, but this time to the Jenkins (request is `https://casita.melchor900.me`):

```
	GET / HTTP/1.1
	Host: pi:8080
	User-Agent: ...
	Accept: */*
	X-Forwarded-For: x.x.x.x
	X-Forwarded-Host: pi:8080
	X-Forwarded-Port: 8080
	X-Forwarded-Proto: https
	X-Forwarded-Server: 4dd420e65da9
	X-Real-Ip: x.x.x.x
	Accept-Encoding: gzip
```

In this case, the `X-Forwarded-Host` is not `casita.melchor9000.me`, is `pi:8080`.

I also tried to use the IP instead of the local domain in the backend url, but the results are the same.

### What did you expect to see?

`X-Forwarded-Host` to be the real one from the original `Host` header.

### What did you see instead?

The host used in the backend configuration.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.6.5
Codename:     tetedemoine
Go version:   go1.10.3
Built:        2018-07-10_03:54:03PM
OS/Arch:      linux/arm
```

**Note**: traefik from `arm32v6/traefik:alpine` docker container.

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```json
{
 "LifeCycle": {
  "RequestAcceptGraceTimeout": 0,
  "GraceTimeOut": 0
 },
 "GraceTimeOut": 0,
 "Debug": true,
 "CheckNewVersion": true,
 "SendAnonymousUsage": false,
 "AccessLogsFile": "",
 "AccessLog": null,
 "TraefikLogsFile": "",
 "TraefikLog": null,
 "Tracing": null,
 "LogLevel": "INFO",
 "EntryPoints": {
  "http": {
   "Address": "xxxx",
   "TLS": null,
   "Redirect": null,
   "Auth": null,
   "WhitelistSourceRange": null,
   "WhiteList": null,
   "Compress": false,
   "ProxyProtocol": null,
   "ForwardedHeaders": null
  },
  "https": {
   "Address": "xxxx",
   "TLS": {
    "MinVersion": "",
    "CipherSuites": null,
    "Certificates": null,
    "ClientCAFiles": null,
    "ClientCA": {
     "Files": null,
     "Optional": false
    }
   },
   "Redirect": null,
   "Auth": null,
   "WhitelistSourceRange": null,
   "WhiteList": null,
   "Compress": true,
   "ProxyProtocol": null,
   "ForwardedHeaders": null
  },
  "panel": {
   "Address": "",
   "TLS": null,
   "Redirect": null,
   "Auth": {
    "Basic": {
     "Users": [],
     "UsersFile": ""
    },
    "Digest": null,
    "Forward": null,
    "HeaderField": ""
   },
   "WhitelistSourceRange": null,
   "WhiteList": null,
   "Compress": false,
   "ProxyProtocol": null,
   "ForwardedHeaders": null
  },
  "traefik": {
   "Address": "xxxx",
   "TLS": null,
   "Redirect": null,
   "Auth": null,
   "WhitelistSourceRange": null,
   "WhiteList": null,
   "Compress": false,
   "ProxyProtocol": null,
   "ForwardedHeaders": null
  }
 },
 "Cluster": null,
 "Constraints": [],
 "ACME": {
  "Email": "xxxx",
  "Domains": [],
  "Storage": "xxxx",
  "StorageFile": "",
  "OnDemand": false,
  "OnHostRule": false,
  "CAServer": "",
  "EntryPoint": "xxxx",
  "DNSChallenge": null,
  "HTTPChallenge": null,
  "DNSProvider": "",
  "DelayDontCheckDNS": 0,
  "ACMELogging": false,
  "TLSConfig": null
 },
 "DefaultEntryPoints": [
  "http",
  "https"
 ],
 "ProvidersThrottleDuration": 2000000000,
 "MaxIdleConnsPerHost": 200,
 "IdleTimeout": 0,
 "InsecureSkipVerify": false,
 "RootCAs": null,
 "Retry": null,
 "HealthCheck": {
  "Interval": 30000000000
 },
 "RespondingTimeouts": null,
 "ForwardingTimeouts": null,
 "AllowMinWeightZero": false,
 "Web": null,
 "Docker": {
  "Watch": true,
  "Filename": "",
  "Constraints": null,
  "Trace": false,
  "TemplateVersion": 0,
  "DebugLogGeneratedTemplate": false,
  "Endpoint": "xxxx",
  "Domain": "xxxx",
  "TLS": null,
  "ExposedByDefault": false,
  "UseBindPortIP": false,
  "SwarmMode": false
 },
 "File": {
  "Watch": true,
  "Filename": "",
  "Constraints": null,
  "Trace": false,
  "TemplateVersion": 0,
  "DebugLogGeneratedTemplate": false,
  "Directory": "/etc/traefik/config/",
  "TraefikFile": ""
 },
 "Marathon": null,
 "Consul": null,
 "ConsulCatalog": null,
 "Etcd": null,
 "Zookeeper": null,
 "Boltdb": null,
 "Kubernetes": null,
 "Mesos": null,
 "Eureka": null,
 "ECS": null,
 "Rancher": null,
 "DynamoDB": null,
 "ServiceFabric": null,
 "Rest": null,
 "API": {
  "EntryPoint": "traefik",
  "Dashboard": true,
  "Debug": false,
  "CurrentConfigurations": null,
  "Statistics": null
 },
 "Metrics": null,
 "Ping": null,
 "ConfigFile": "/etc/traefik/traefik.toml"
}
```

File `/etc/traefik/config/jenkins.toml`:
```
[frontends]
    [frontends.jenkins]
    backend = "jenkinsPi"
        [frontends.jenkins.routes.test_1]
        rule = "Host:casita.melchor9000.me"
        [frontends.jenkins.redirect]
        entryPoint = "https"
        permanent = true
        [frontends.jenkins.headers]
        SSLRedirect = true

[backends]
    [backends.jenkinsPi]
        [backends.jenkinsPi.servers.pi]
        url = "http://pi:8080"
```

**Comment by melchor629** [comment]
After a bit of debugging today, I found what the issue is. It turn out to be that `passHostHeader` setting for frontends is set to `false` by default, but in the [documentation][1] suggests that the default is `true`. Docker provider uses `true` by default, for example.

Maybe, to fix this, should be set to `true` by default when reading the configuration from files, or change the documentation to set it to `false`, so no one could get this trouble again.

My `jenkins.toml` file now looks like:

```
[frontends]
    [frontends.jenkins]
    backend = "jenkinsPi"
    passHostHeader = true # <-Added
        [frontends.jenkins.routes.test_1]
        rule = "Host:casita.melchor9000.me"
        [frontends.jenkins.redirect]
        entryPoint = "https"
        permanent = true
        [frontends.jenkins.headers]
        SSLRedirect = true

[backends]
    [backends.jenkinsPi]
        [backends.jenkinsPi.servers.pi]
        url = "http://pi:8080"
```

...and works as expected.

  [1]: https://docs.traefik.io/configuration/backends/file/

**Comment by traefiker** [comment]
Closed by #3777.
